qiskit/extensions/standard/u1.py

```diff
@@ -340,7 +242,72 @@ def mcu1(self, lam, control_qubits, target_qubit):
             circuit.draw()
     """
     num_ctrl_qubits = len(control_qubits)
-    return self.append(MCU1Gate(lam, num_ctrl_qubits), control_qubits[:] + [target_qubit], [])
+    return self.append(CU1Gate(lam, num_ctrl_qubits), control_qubits[:] + [target_qubit], [])
 
 
 QuantumCircuit.mcu1 = mcu1
+
+
+def _mcu1_rule(q, lam, num_ctrl_qubits):
+    """The gate definition of the multi-controlled U1 gate.
+
+    Ported from Aqua (github.com/Qiskit/qiskit-aqua),
+    commit 769ca8d, file qiskit/aqua/circuits/gates/multi_control_u1_gate.py.
+    """
+    rule = []
+    q_controls = q[:num_ctrl_qubits]
+    q_target = q[num_ctrl_qubits]
+    if num_ctrl_qubits == 0:
+        rule.append(
+            (U1Gate(lam), [q_target], [])
+        )
+    elif num_ctrl_qubits == 1:
+        rule.append(
+            (CU1Gate(lam), q_controls + [q_target], [])
+        )
+    else:
+        from qiskit.extensions.standard.x import CXGate
+        from sympy.combinatorics.graycode import GrayCode
```

Comment:
Can you use `_generate_gray_code` from `multi_control_rotation_gates.py` instead?

Reply:
Is it preferred to use 'our' version compared to the one from sympy? I thought it is better to use the function from a standardized library. I actually wanted to remove `_generate_gray_code` with the implementations of the multi controlled Pauli rotations.